### PR TITLE
Fix coordinator selection

### PR DESCRIPTION
**Pull request by johscheuer**
Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/713

### CODE REVIEW

File: controllers/change_coordinators_test.go [hunk 1]
```diff
@@ -40,79 +42,285 @@ var _ = Describe("Reconcile.SelectCandidates", func() {
...
-		Context("with candidates", func() {
-			var candidates []localityInfo
-			JustBeforeEach(func() {
-				candidates = make([]localityInfo, 0, len(status.Cluster.Processes))
+			candidates, err = selectCoordinators(cluster, status)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("all processes are healthy", func() {
+			FIt("should only select storage processes in primary and remote and Tlog in satellite", func() {
```
Comment (brownleej) [inline controllers/change_coordinators_test.go:171]:
```suggestion
			It("should only select storage processes in primary and remote and Tlog in satellite", func() {
```

File: controllers/change_coordinators_test.go [hunk 2]
```diff
@@ -40,79 +42,285 @@ var _ = Describe("Reconcile.SelectCandidates", func() {
...
+
+			// generate status for 2 dcs and 1 sate
+			status.Cluster.Processes = generateProcessInfo(2, 2)
+
+			candidates, err = selectCoordinators(cluster, status)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("all processes are healthy", func() {
+			FIt("should only select storage processes in primary and remote and Tlog in satellites", func() {
```
Comment (brownleej) [inline controllers/change_coordinators_test.go:229]:
```suggestion
			It("should only select storage processes in primary and remote and Tlog in satellites", func() {
```

File: controllers/change_coordinators_test.go (hunk 1 above)
Comment (johscheuer) [reply controllers/change_coordinators_test.go:171]: Yeah, I still have to add some more tests to ensure the selection works also in some other weird cases or return the correct error. I actually thought I marked thee PR as draft 🤔

**Review by brownleej — APPROVED** [review]
👍

**Pull request merged**